src/components/HomeVideoGrid/HomeVideoGrid.test.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { act, render } from '@testing-library/react';
+import api from 'api';
+import { ProviderWrapper } from 'test';
+import HomeVideoGrid from './HomeVideoGrid.component';
+
+jest.mock('api');
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
+  useParams: () => ({
+    videoId: 'video1',
+  }),
+}));
+const videoData = [
+  {
+    etag: '1',
+    id: { videoId: 'video1' },
+    snippet: {
+      title: 'Title',
+      description: 'This is the description',
+      thumbnails: {
+        medium: {
+          url: 'http://theurl.com',
+        },
+      },
+    },
+  },
+  {
+    etag: '2',
+    id: { videoId: 'video2' },
+    snippet: {
+      title: 'Title 2',
+      description: 'This is the description 2',
+      thumbnails: {
+        medium: {
+          url: 'http://theurl2.com',
+        },
+      },
+    },
+  },
+];
```

Comment:
I believe you could extract the repeated code, insert it into a file and, import the file wherever you need to mock those dependencies, you can also export the videoData array in the same file. 
```js
// mock-dependencies.js
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    videoId: 'video1',
  }),
}));
export const videoData = [...];

// HomeVideoGrid.test.jsx
import { videoData } from 'path_to_file/mock-dependencies.js'
```
Or you could create a setup file for all the test suites where it is executed once per file [Jest documentation](https://jestjs.io/docs/configuration#setupfiles-array)